File: UClient/Assets/UFramework/Scripts/Service/Core/BaseService.cs
/*
 * @Author: fasthro
 * @Date: 2020-05-25 23:52:22
 * @Description: BaseManager
 */

using System;
using LuaInterface;
using UnityEngine;

namespace UFramework
{
    public abstract class BaseService : BaseUnityBehaviourBindLua
    {
        public BaseService()
        {
            Initialize();
        }
    }
}

File: UClient/Assets/Scripts/Lua/Service/BaseService.lua
--[[
Author: fasthro
Date: 2020-12-28 12:09:16
Description: 
--]]
local BaseService =
    typesys.def.BaseService {
    _ext = typesys.__unmanaged
}

function BaseService:__ctor(ext)
    self._ext = ext

    if ext ~= nil then
        ext.luaTable = self
    end
end

function BaseService:__dtor()
end

function BaseService:initialize()
end

return BaseService